src/App.jsx

```diff
@@ -21,6 +21,19 @@ const App = () => {
     setName(newName)
   }
 
+  useEffect(() => {
+    window.addEventListener("visibilitychange", onVisibilityChange);
+    return () => {
+      window.removeEventListener("visibilitychange", onVisibilityChange);
+    };
+  });
+
+  const onVisibilityChange = () => {
+    if (document.visibilityState === "visible") {
+      navigator.clearAppBadge()
+    }
+  }
+
```

Comment:
could you move this into the same component where you call `navigator.setAppBadge()`?